src/App.tsx

```diff
@@ -1,61 +1,141 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { FormEvent, useState } from 'react';
+
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const getId = (todosList: Todo[]): number => {
+  return Math.max(...todosList.map(todo => todo.id)) + 1;
+};
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setUserId] = useState(0);
+  const [todosList, setTodos] = useState(todos);
+  const [showTitleError, setShowTitleError] = useState(false);
+  const [showUserError, setShowUserError] = useState(false);
+
+  const clear = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    const trimmedTitle = title.trim();
+
+    const isValid = trimmedTitle.replace(
```

Comment:
Consider using a more descriptive variable name instead of `isValid`. For example, `isTitleValid`. 🤔